File: Python/leetcode_933.py
from collections import deque

class RecentCounter:
    def __init__(self):
      self.requests = deque()

    def ping(self, t: int) -> int:
      self.requests.append(t)
      if len(self.requests) == 1:
        return 1
      else:
        while True:
          first = self.requests[0]
          last = self.requests[-1]
          
          if (last - first) > 3000:
            self.requests.popleft()
          else:
            return len(self.requests)
        


# Your RecentCounter object will be instantiated and called as such:
obj = RecentCounter()
print(obj.ping(1))
print(obj.ping(100))
print(obj.ping(200))
print(obj.ping(3000))
print(obj.ping(3010))
print(obj.ping(3020))

File: JavaScript/leetcode_933.ts
class RecentCounter {
  requests: number[];
  pointerOne: number;
  count: number;

  constructor() {
    this.count = 0;
    this.pointerOne = 0;
    this.requests = new Array();
  }

  ping(t: number): number {
    this.requests.push(t);
    this.count++;
    if (this.count == 1) return 1;
    while (true) {
      let first = this.requests[0];
      let last = this.requests[this.count - 1];

      if (last - first > 3000) {
        this.requests.shift();
        this.count--;
      } else {
        return this.count;
      }
    }
  }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * var obj = new RecentCounter()
 * var param_1 = obj.ping(t)
 */

var obj = new RecentCounter();
console.log(obj.ping(1));
console.log(obj.ping(100));
console.log(obj.ping(200));
console.log(obj.ping(3000));
console.log(obj.ping(3010));
console.log(obj.ping(3020));
